elements/layercontrol/src/components/layer.js

```diff
@@ -43,6 +52,21 @@ export class EOxLayerControlLayer extends LitElement {
     this.unstyled = false;
   }
 
+  /**
+   * Run zoom event to zoom level change
+   * and update DOM if `showZoomLayerState` present in layer property
+   */
+  updated() {
+    if (this.layer?.get("showZoomLayerState")) {
+      this.map.getView().on("change:resolution", () => {
```

Comment:
Registering the listener for `change:resolution` each time the element updates (`updated()`), the listeners actually stack and the more the element it uses, the more time this re-rendering triggers.

Adding a console logs in e.g. lines 61 and 64 shows this issue, which seems to slow down the website after zooming back and forth a bit and changing changing layer order:

![image](https://github.com/EOX-A/EOxElements/assets/26576876/a0ca49e7-cf70-487d-a540-5519b7cb9270)
![image](https://github.com/EOX-A/EOxElements/assets/26576876/5278aa2a-968a-41fb-805f-6f0b86dafd56)

Reply:
@silvester-pari 
I have resolved re-rendering and event stacking issue.
I have used `firstUpdated()` instead of `updated()` to stop stacking of event 

But it will still create event of layer which has `showZoomLayerState: true`. We can still use one event function by putting everything inside `layerList` and controlling it with a list of zoom layer states but then it won't make sense because the functionality should be part of `layer` not it's list.